### ISSUE TITLE
Constant sprintf() calls

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#2183]
I see the following unnecessary `sprintf()` across the codebase I maintain:

```r
sprintf("hello")

sprintf("%s", "hello")
```

In other words:

- `sprintf()` does not have any arguments passed to `...` 
- `sprintf()` `fmt` is `"%s"` **and** the next argument in a `STR_CONST`. 

These two examples could be replaced by a more readable and more efficient `"hello"`.

I believe the existing `sprintf_linter()` infrastructure and in particular the logic around piped calls could be re-used here. If you believe this would be a good improvement, I can give it a try.

**Comment by AshesITR** [comment]
I think `sprintf_linter()` is a good place (and `paste_linter()` for the other case. There is no coercion going on in `sprintf("%s", "a")`.

**Comment by MichaelChirico** [comment]
Large number of hits on GitHub:

https://github.com/search?q=%2F%5B%5Ea-z0-9._%5Dsprintf%5B%28%5D%5B%22%27%5D%5B%5E%27%22%5D*%5B%27%22%5D%5B%29%5D%2F+lang%3AR&type=code

**Comment by MichaelChirico** [comment]
There's some subtlety here, it's not quite trivial to implement:

```r
sprintf("%s")         # already lints b/c '%s' is not matched
sprintf("abc")        # needs to lint because there's no template at all
sprintf("def", "ghi") # already lints b/c 'ghi' doesn't correspond to anything
```

i.e. it's not as simple as "find `sprintf()` calls with just one argument", it's got to first subset among those calls that don't already generate a lint.

Not particularly difficult, but requires a bit more care.

**Comment by AshesITR** [comment]
Another edge case to be aware of is `sprintf("%%")` and `sprintf("%%", "")`. The former could lint (replace by `"%"`). The latter already lints.

`contains(text(), '%')` is too simple.
